src/tests/dummytask.py

```diff
@@ -0,0 +1,59 @@
+"""
+Copyright 2020 Google LLC.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import time
+import random
+import string
+import threading
+LETTERS = string.ascii_lowercase
+STATES = ["READY", "RUNNING", "COMPLETED"]
+
+
+class DummyTask:
+    """Dummy Task class to emulate EE tasks."""
```

Comment:
This isn't really emulating an EE task right? It loops through the states until cancelled, and its state is not deterministic. 

I'm assuming this was done to allow easier testing, for which you should use https://docs.python.org/3/library/threading.html#barrier-objects to synchronize your test objects.